app/modules/annotations/resources.py

```diff
@@ -118,6 +115,37 @@ def post(self, args):
         return annotation
 
 
+@api.route('/search')
+@api.login_required(oauth_scopes=['annotations:read'])
+class AnnotationElasticsearch(Resource):
+    @api.permission_required(
+        permissions.ModuleAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'module': Annotation,
+            'action': AccessOperation.READ,
+        },
+    )
+    @api.response(schemas.BaseAnnotationSchema(many=True))
+    @api.paginate()
+    def get(self, args):
+        search = {}
+        return Annotation.elasticsearch(search, **args)
+
+    @api.permission_required(
+        permissions.ModuleAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'module': Annotation,
+            'action': AccessOperation.READ,
+        },
+    )
+    @api.response(schemas.BaseAnnotationSchema(many=True))
+    @api.paginate()
+    def post(self, args):
+        search = request.get_json()
```

Comment:
Please can you explain the reasoning behind having both GET and POST with the only difference being this one line. Why can't they be one API?

Reply:
The new APIs for searching a module with Elasticsearch use the HTTP `POST` method because we need to allow the user to specify a query body to Elasticsearch.  This is not possible to do with `GET` because that method does not allow FormData to be sent as JSON.  The benefit of adding an... let's call it an "alias" on `GET`... is that it is very handy when using the browser to simply list objects from Elasticsearch.  We offer the `POST` API as the official API where users can search for things and the `GET` API to simply get all search items out of Elasticsearch with an assumed query body of `{}`.  This is mostly handy for debugging and to exist as a complement to the `/api/v1/<module>/` listing APIs that support very basic filtering.

There may be a way to prevent some of the duplication, but I wanted to keep the decorators clean and standardized.  I left the two functions as stand-alone functions because I don't want to muddy the waters with more complicated or custom security/access.  I went with adding a convenience API with explicit decorators, at the cost of some duplicated boilerplating.